### PR TITLE
OOP school library: unit tests

### PR DESCRIPTION
**Pull request by MuhammadUsmanAslam**
The following changes are being requested to merge:

- Add unit tests for all the methods in all the classes:

   - Book
   - Person
   - Student
   - Teacher
   - Classroom
   - Rental

### CODE REVIEW

**Review by julie-ify — COMMENTED** [review]
Hi @MuhammadUsmanAslam and @ikoote1,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### For more professional documentation, kindly remove the Live demo section and its link on the menu since you won't be deploying the app, according to you.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      expect(person.correct_name).to eq('Usman')
+    end
+  end
+
+  describe 'can_use_services' do
+    it 'returns true if the person can use services' do
+      expect(person.can_use_services?).to equal(true)
+    end
+  end
+end
```
Comment (julie-ify) [inline spec/person_spec.rb]: - [x] Kindly add more Person class tests for more attributes, like name, age, parent_permission, rentals and the `rent_book` method.

- [x] Kindly add tests for the classes you created while creating the Decorator pattern. The classes include nameable, decorator, capitalize and trimmer.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @MuhammadUsmanAslam and @ikoote1,

Good job so far!

## Highlights!
- Good code organization and tests ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by youmari — APPROVED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @MuhammadUsmanAslam @ikoote1 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**